pkg/awsvpc/analysis_output_test.go

```diff
@@ -0,0 +1,426 @@
+	format vpcmodel.OutFormat,
+	configName string,
+	allVPCs bool) (
+	expectedFileName,
+	actualFileName string,
+	err error) {
+	var res string
+
+	// if there are more than one vpc in the config, split to a file per one vpc analysis
+	baseName := testName
+	if allVPCs {
+		baseName += "_all_vpcs_"
+	} else {
+		baseName += "_" + configName
+	}
+
+	switch uc {
+	case vpcmodel.AllEndpoints:
+		res = baseName
+	case vpcmodel.SingleSubnet:
+		res = baseName + suffixOutFileDebugSubnet
+	case vpcmodel.AllSubnets:
+		res = baseName + suffixOutFileSubnetsLevel
+	case vpcmodel.AllSubnetsNoPGW:
+		res = baseName + suffixOutFileSubnetsLevelNoPGW
+	case vpcmodel.SubnetsDiff:
+		res = baseName + suffixOutFileDiffSubnets
+	case vpcmodel.EndpointsDiff:
+		res = baseName + suffixOutFileDiffEndpoints
+	case vpcmodel.Explain:
+		res = baseName + suffixOutFileExplain
+	}
+	if grouping {
+		res += suffixOutFileWithGrouping
+	}
+	if noLbAbstract {
+		res += suffixOutFileWithoutLbAbstraction
+	}
+	suffix, suffixErr := getTestFileSuffix(format)
+	if suffixErr != nil {
+		return "", "", suffixErr
+	}
+	res += suffix
+
+	expectedFileName = res
+	actualFileName = actualOutFilePrefix + res
+	return expectedFileName, actualFileName, nil
+}
+
+func getTestFileSuffix(format vpcmodel.OutFormat) (suffix string, err error) {
+	switch format {
+	case vpcmodel.Text:
+		return txtOutSuffix, nil
+	case vpcmodel.Debug:
+		return debugOutSuffix, nil
+	case vpcmodel.MD:
+		return mdOutSuffix, nil
+	case vpcmodel.JSON:
+		return jsonOutSuffix, nil
+	case vpcmodel.DRAWIO:
+		return drawioOutSuffix, nil
+	case vpcmodel.ARCHDRAWIO:
+		return archDrawioOutSuffix, nil
+	case vpcmodel.SVG:
+		return svgOutSuffix, nil
+	case vpcmodel.ARCHSVG:
+		return archSvgOutSuffix, nil
+	case vpcmodel.HTML:
+		return htmlOutSuffix, nil
+	case vpcmodel.ARCHHTML:
+		return archHTMLOutSuffix, nil
+	default:
+		return "", errors.New("unexpected out format")
+	}
+}
+
+// initTest: based on the test name, set the input config file name, and the output
+// files names (actual and expected), per use case
+func (tt *vpcGeneralTest) initTest() {
+	tt.inputConfig2nd = inputFilePrefix + tt.inputConfig + secJSONOutSuffix
+	tt.inputConfig = inputFilePrefix + tt.inputConfig + jsonOutSuffix
+	tt.expectedOutput = map[vpcmodel.OutputUseCase]string{}
+	tt.actualOutput = map[vpcmodel.OutputUseCase]string{}
+	// init field of expected errs
+	if tt.errPerUseCase == nil {
+		tt.errPerUseCase = map[vpcmodel.OutputUseCase]error{}
+	}
+	for _, uc := range tt.useCases {
+		if _, ok := tt.errPerUseCase[uc]; !ok {
+			tt.errPerUseCase[uc] = nil // if errs not specified, by default not expecting errs
+		}
+	}
+}
+
+var tests = []*vpcGeneralTest{
+	{
+		inputConfig: "basic_config_with_sg",
+		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints},
+		format:      vpcmodel.Text,
+	},
+	{
+		inputConfig: "aws_sg_1",
+		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints},
+		format:      vpcmodel.Text,
```

Comment:
can add some details in comment about non trivial test details?
why are the SG rules with not "all allowed" connections, not reflected in the expected connectivity report?

Reply:
I meant to `SG GroupId:20` with rules where not all protocols are permitted. It seems they have no effect because their IP ranges are empty. I'd consider changing that.

Reply:
changed